src/module.c

```diff
@@ -289,68 +277,28 @@ int RSCursorCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc);
  *  Delete a document from the index. Returns 1 if the document was in the index, or 0 if not.
  *
  *  **NOTE**: This does not actually delete the document from the index, just marks it as deleted
- * If DD (Delete Document) is set, we also delete the document.
+ *  If DD (Delete Document) is set, we also delete the document.
+ *  Since v2.0, document is deleted by default.
  */
 int DeleteCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   RedisModule_AutoMemory(ctx);
 
-  if (argc < 3 || argc > 4) return RedisModule_WrongArity(ctx);
+  if (argc != 3) return RedisModule_WrongArity(ctx);
   IndexSpec *sp = IndexSpec_Load(ctx, RedisModule_StringPtrLen(argv[1], NULL), 1);
   if (sp == NULL) {
     return RedisModule_ReplyWithError(ctx, "Unknown Index name");
   }
 
-  int delDoc = 0;
-  if (argc == 4 && RMUtil_StringEqualsCaseC(argv[3], "DD")) {
-    delDoc = 1;
+  RedisModuleCallReply *rep = NULL;
+  RedisModuleString *doc_id = argv[2];
+  rep = RedisModule_Call(ctx, "DEL", "s", doc_id);
+  if (rep == NULL || RedisModule_CallReplyType(rep) != REDISMODULE_REPLY_INTEGER || 
+      RedisModule_CallReplyInteger(rep) != 1) {
+    return RedisModule_ReplyWithError(ctx, "Document was not found");
```

Comment:
RedisModule_ReplyWithCallReply?

Reply:
`return RedisModule_ReplyWithLongLong(ctx, 0)`?
This will be more consistent with previous

Reply:
Why not expose the error returned from rm_call?